maths/kadanes_algorithm.py

```diff
@@ -0,0 +1,61 @@
+"""
+    Link for the explanation of this algorithm :- 
+    https://medium.com/@rsinghal757/kadanes-algorithm-dynamic-programming-how-and-why-does-it-work-3fd8849ed73d
+    Kadane's algorithm to get maximum subarray sum
+    Please enter the decimal values only and use spaces as separator
+"""
+def negative_exist(arr):

```

Comment:
```suggestion
test_data = ([-2, -8, -9], [2, 8, 9], [-1, 0, 1], [0, 0], [])


def negative_exist(arr: list) -> int:
```
Different lists: all negative, all positive, mixed, all zero, empty list.